admin/load.php

```diff
@@ -544,10 +547,23 @@ function perflab_activate_plugin() {
 		return;
 	}
 
-	// The plugin being activated.
 	$plugin = sanitize_text_field( wp_unslash( $_GET['plugin'] ) );
 
-	check_admin_referer( "perflab_activate_plugin_{$plugin}" );
+	// Check whether "plugin" is a plugin slug or a plugin basename.
+	if ( str_contains( $plugin, '/' ) ) {
+		check_admin_referer( "perflab_activate_plugin_{$plugin}" );
+	} else {
+		check_admin_referer( "perflab_activate_plugin_{$plugin}" );
+
```

Comment:
This can be simplified as the admin referer check remains the same regardless of whether it's a plugin slug or basename. The respective logic dealing with the nonce handles this for both approaches.
```suggestion
	check_admin_referer( "perflab_activate_plugin_{$plugin}" );

	// If `$plugin` is a plugin slug rather than a plugin basename, determine the full plugin basename.
	if ( ! str_contains( $plugin, '/' ) ) {
```